### ISSUE TITLE
maximum [] 

### ISSUE THREAD

**Issue by adamflott** [sdiehl/protolude#47]
Caught this while my code (which relies on protoludes banishes partial functions) was in QA. I'm not sure if changing the type to match `head` behavior is desired, so it's up to you what to do. Wanted it known.

```
15:43:42 (19:43) [@bos-lp6rc:~/src/protolude] master* ± stack ghci
protolude-0.2: configure (lib)
Configuring protolude-0.2...
protolude-0.2: initial-build-steps (lib)
Configuring GHCi with the following packages: protolude
GHCi, version 8.0.1: http://www.haskell.org/ghc/  :? for help
Loaded GHCi configuration from /home/aflott/src/protolude/.ghci

<no location info>: error:
    Could not find module ‘Protolude’
    It is not a module in the current program, or in any known package.
Loaded GHCi configuration from /home/aflott/.ghci
[ 1 of 16] Compiling Bifunctor        ( /home/aflott/src/protolude/src/Bifunctor.hs, interpreted )
[ 2 of 16] Compiling Semiring         ( /home/aflott/src/protolude/src/Semiring.hs, interpreted )
[ 3 of 16] Compiling List             ( /home/aflott/src/protolude/src/List.hs, interpreted )
[ 4 of 16] Compiling Base             ( /home/aflott/src/protolude/src/Base.hs, interpreted )
[ 5 of 16] Compiling Conv             ( /home/aflott/src/protolude/src/Conv.hs, interpreted )
[ 6 of 16] Compiling Exceptions       ( /home/aflott/src/protolude/src/Exceptions.hs, interpreted )
[ 7 of 16] Compiling Unsafe           ( /home/aflott/src/protolude/src/Unsafe.hs, interpreted )
[ 8 of 16] Compiling Either           ( /home/aflott/src/protolude/src/Either.hs, interpreted )
[ 9 of 16] Compiling Applicative      ( /home/aflott/src/protolude/src/Applicative.hs, interpreted )
[10 of 16] Compiling Functor          ( /home/aflott/src/protolude/src/Functor.hs, interpreted )
[11 of 16] Compiling Monad            ( /home/aflott/src/protolude/src/Monad.hs, interpreted )
[12 of 16] Compiling Bool             ( /home/aflott/src/protolude/src/Bool.hs, interpreted )
[13 of 16] Compiling Show             ( /home/aflott/src/protolude/src/Show.hs, interpreted )
[14 of 16] Compiling Panic            ( /home/aflott/src/protolude/src/Panic.hs, interpreted )
[15 of 16] Compiling Debug            ( /home/aflott/src/protolude/src/Debug.hs, interpreted )
[16 of 16] Compiling Protolude        ( /home/aflott/src/protolude/src/Protolude.hs, interpreted )
Ok, modules loaded: Debug, Panic, Show, Bool, Monad, Functor, Applicative, Either, Protolude, Unsafe, Base, List, Semiring, Conv, Exceptions, Bifunctor.
Loaded GHCi configuration from /tmp/ghci29773/ghci-script
*Unsafe Prelude Applicative Base Bifunctor Bool Conv Debug Either Exceptions Functor List Monad Panic Protolude Semiring Show Unsafe>
*Unsafe Prelude Applicative Base Bifunctor Bool Conv Debug Either Exceptions Functor List Monad Panic Protolude Semiring Show Unsafe>
*Unsafe Prelude Applicative Base Bifunctor Bool Conv Debug Either Exceptions Functor List Monad Panic Protolude Semiring Show Unsafe> maximum []

<interactive>:3:1: warning: [-Wtype-defaults]
    • Defaulting the following constraints to type ‘()’
        (Ord a0) arising from a use of ‘it’ at <interactive>:3:1-10
        (Show a0)
          arising from a use of ‘Base.print’ at <interactive>:3:1-10
    • In the first argument of ‘Base.print’, namely ‘it’
      In a stmt of an interactive GHCi command: Base.print it
*** Exception: Prelude.maximum: empty list
```
I expected Nothing like `head`:

```
*Unsafe Prelude Applicative Base Bifunctor Bool Conv Debug Either Exceptions Functor List Monad Panic Protolude Semiring Show Unsafe> head []

<interactive>:4:1: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘()’
        Show a0 arising from a use of ‘Base.print’
    • In a stmt of an interactive GHCi command: Base.print it
Nothing

```

**Comment by sdiehl** [comment]
I think you want `maximumMay`: https://hackage.haskell.org/package/safe-0.3.14/docs/Safe.html#v:maximumMay

**Comment by sdiehl** [comment]
Fixed in 456a063.
